lib/stripe.rb

```diff
@@ -88,6 +90,12 @@ def self.request(method, url, api_key, params={}, headers={}, api_base_url=nil)
         'email support@stripe.com if you have any questions.)')
     end
 
+    if method == :get && headers.key?(:idempotency_key)
+      raise ArgumentError.new(
+        'Idempotency keys cannot be used for GET requests.'
```

Comment:
GET requests are idempotent by nature. Seems a bit harsh to raise here instead of just ignoring